cmd/kogito/command/context/config.go

```diff
@@ -46,36 +47,44 @@ type Configuration struct {
 func InitConfig() {
 	if rootCmd.ConfigFile() != "" {
 		// Use config file from the flag.
-		viper.SetConfigFile(rootCmd.ConfigFile())
+		fullPath, file := filepath.Split(rootCmd.ConfigFile())
+		updateConfigFile(fullPath, file)
 	} else {
 		// Find home directory.
 		home, err := homedir.Dir()
 		if err != nil {
 			fmt.Println(err)
 			os.Exit(1)
 		}
-		// config viper
+		// Setup full path
 		fullPath := filepath.Join(home, DefaultConfigPath)
-		viper.AddConfigPath(fullPath)
-		viper.SetConfigName(DefaultConfigFile)
-		viper.SetConfigType(DefaultConfigExt)
-		// ensure file
-		if err := viper.ReadInConfig(); err != nil {
-			if _, ok := err.(viper.ConfigFileNotFoundError); ok {
-				if err := os.MkdirAll(fullPath, os.ModePerm); err != nil {
-					panic(fmt.Errorf("Error creating path for config file: %s ", err))
-				} else {
-					if err := viper.WriteConfigAs(filepath.Join(fullPath, DefaultConfigFinalName)); err != nil {
-						panic(fmt.Errorf("Error while trying to write config file: %s ", err))
-					}
-				}
+		updateConfigFile(fullPath, DefaultConfigFinalName)
+	}
+
+	viper.AutomaticEnv() // read in environment variables that match
+}
+
+func updateConfigFile(fullPath, file string) {
```

Comment:
Have you tested the built CLI with the `--config` flag? I mean something like `kogito use-project kogito --config /tmp/myfile.yaml`?

Reply:
tested and approved, with and without the --config flag. Config file is correctly used :)